internal/filegreeter/greeter_test.go

```diff
@@ -0,0 +1,41 @@
+package filegreeter
+
+import (
+	"bytes"
+	"context"
+	"testing"
+)
+
+func TestFileGreeter_Greet(t *testing.T) {
+	var buffer bytes.Buffer
+	buffer.WriteString("jimmy")
+	content := readFile(&buffer)
+	tests := []struct {
+		name   string
+		in     string
+		expect string
+	}{
+		{
+			name:   "Happy - Name matches file",
+			in:     "jimmy",
+			expect: "Hello, jimmy!",
+		},
+		{
+			name:   "Sad - Name does not match file",
+			in:     "jim",
+			expect: "Hello, World!",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := &FileGreeter{
+				file:    &buffer,
+				content: content,
+			}
+			if got := g.Greet(context.Background(), tt.in); got != tt.expect {
```

Comment:
while this works in simple tests like this, check out the `testify/assert` package for more flexible checks